### PR TITLE
core: fix ASLR problem with short table mappings

### PR DESCRIPTION
**Pull request by jenswi-linaro**
With short table mappings, that is without LPAE, the user va range is
defined at the lowest addresses. Depending on the seed supplied this
could conflict with chosen base address for core mappings. Add a check
early in assign_mem_va() to avoid such conflicts.

Without this patch there's a risk of occasional panics like:
E/TC:0 0 Panic 'issue in linear address space' at core/arch/arm/mm/core_mmu.c:2147 <check_pa_matches_va>
E/TC:0 0 TEE load address @ 0xa34000
E/TC:0 0 Call stack:
E/TC:0 0  0x00a3a901

Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>

### CODE REVIEW

File: core/arch/arm/mm/core_mmu.c [hunk 1]
```diff
@@ -916,6 +916,19 @@ static bool assign_mem_va(vaddr_t tee_ram_va,
 	vaddr_t va = tee_ram_va;
 	bool va_is_secure = true;
 
+	/*
+	 * Check that we're not overlapping with the user va range in case
+	 * we're not using LPAE.
```
[the remark this answers is not in the document]
Comment (jenswi-linaro) [reply core/arch/arm/mm/core_mmu.c]: I'll update to make it more clear.

**Comment by jenswi-linaro** [comment]
Update

File: core/arch/arm/mm/core_mmu.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (jforissier) [reply core/arch/arm/mm/core_mmu.c]: Thanks. That's a difference I was not aware of. LGTM.

**Comment by jenswi-linaro** [comment]
Squashed and tag applied.

File: core/arch/arm/mm/core_mmu.c [hunk 2]
```diff
@@ -916,6 +916,25 @@ static bool assign_mem_va(vaddr_t tee_ram_va,
 	vaddr_t va = tee_ram_va;
 	bool va_is_secure = true;
 
+	/*
+	 * Check that we're not overlapping with the user VA range.
+	 */
+	if (IS_ENABLED(CFG_WITH_LPAE)) {
+		/*
+		 * User VA range is supposed to be defined after these
+		 * mappings has been established.
```
Comment (etienne-lms) [inline core/arch/arm/mm/core_mmu.c:925]: s/has/have/

**Review by etienne-lms — COMMENTED** [review]
Sorry for late feedback. LGTM for the implementation.
Minor comment on commit log: prefer `32bit MMU` or `short-descriptor table mappings` over `short table mappings`. In Arm litterature, `short` is rather used as `short-descriptor` when referring to non-LPAE mapping.

`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>` with applicable comments addressed.

**Comment by jenswi-linaro** [comment]
Comments addressed, tag applied.

**Pull request merged**